tests/www/news/test_views.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from django.core.cache import cache
+from django.urls import reverse
+
+from itou.communications.cache import CACHE_ACTIVE_ANNOUNCEMENTS_KEY
+from itou.communications.models import AnnouncementCampaign
+from itou.users.enums import UserKind
+from itou.utils.constants import ITOU_HELP_CENTER_URL
+from tests.communications.factories import AnnouncementCampaignFactory
+from tests.users.factories import JobSeekerFactory, PrescriberFactory
+from tests.utils.test import parse_response_to_soup, reset_model_sequence
+
+
+reset_announcement_campaign = pytest.fixture(reset_model_sequence(AnnouncementCampaign))
+
+
+class TestNewsRender:
+    def test_campaign_rendered_dashboard_prescripteur(self, client, snapshot, reset_announcement_campaign):
+        cache.delete(CACHE_ACTIVE_ANNOUNCEMENTS_KEY)
```

Comment:
Peut-être en faire une fixture en autouse pour cette classe ?

Reply:
J’imaginais un truc comme ça, parce que je trouve que la mark est très spécifique, mais peu importe.

```diff
diff --git a/pytest.ini b/pytest.ini
index c73ef3acc..39c432b4e 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -9,4 +9,3 @@ addopts =
     --strict-markers
 markers =
     no_django_db: mark tests that should not be marked with django_db.
-    empty_active_announcements_cache: mark tests that use the active AnnouncementCampaign cache.
diff --git a/tests/communications/test_cache.py b/tests/communications/test_cache.py
index e0a684a9c..544627282 100644
--- a/tests/communications/test_cache.py
+++ b/tests/communications/test_cache.py
@@ -11,8 +11,11 @@ from tests.communications.factories import AnnouncementCampaignFactory, Announce
 from tests.utils.test import TestCase
 
 
-@pytest.mark.empty_active_announcements_cache
 class AnnouncementCampaignCacheTest(TestCase):
+    @pytest.fixture(autouse=True)
+    def empty_announcements_cache(self, empty_active_announcements_cache):
+        pass
+
     def test_active_announcement_campaign_context_processor_cached(self):
         campaign = AnnouncementCampaignFactory(with_item=True)
 
diff --git a/tests/communications/test_campaigns.py b/tests/communications/test_campaigns.py
index 30a037ef0..daca24f94 100644
--- a/tests/communications/test_campaigns.py
+++ b/tests/communications/test_campaigns.py
@@ -74,8 +74,11 @@ class TestAnnouncementCampaignAdmin(TestCase):
         assert response.status_code == 200
 
 
-@pytest.mark.empty_active_announcements_cache
 class TestRenderAnnouncementCampaign:
+    @pytest.fixture(autouse=True)
+    def empty_announcements_cache(self, empty_active_announcements_cache):
+        pass
+
     def test_campaign_rendered_dashboard(self, client, snapshot):
         campaign = AnnouncementCampaignFactory(max_items=3)
         AnnouncementItemFactory(campaign=campaign, title="Item A", description="Item A", priority=0)
diff --git a/tests/conftest.py b/tests/conftest.py
index 697d1a435..afd1317d4 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -148,13 +148,18 @@ def cache_per_test(settings):
     settings.CACHES = caches
 
 
+@pytest.fixture
+def empty_active_announcements_cache():
+    cache.delete(CACHE_ACTIVE_ANNOUNCEMENTS_KEY)
+    yield
+
+
 @pytest.fixture(autouse=True)
 def cached_announce_campaign(request):
     """
     Populates cache for AnnouncementCampaign to avoid an extra database hit in many tests
     """
-    if "empty_active_announcements_cache" not in request.keywords:
-        cache.set(CACHE_ACTIVE_ANNOUNCEMENTS_KEY, None, None)
+    cache.set(CACHE_ACTIVE_ANNOUNCEMENTS_KEY, None, None)
     yield
     cache.delete(CACHE_ACTIVE_ANNOUNCEMENTS_KEY)
 
diff --git a/tests/utils/tests.py b/tests/utils/tests.py
index 694fc42a7..6fdbb4ba6 100644
--- a/tests/utils/tests.py
+++ b/tests/utils/tests.py
@@ -1443,8 +1443,7 @@ def test_job_application_state_badge_oob_swap(snapshot):
     assert job_applications.state_badge(job_application, hx_swap_oob=True) == snapshot
 
 
-@pytest.mark.empty_active_announcements_cache
-def test_active_announcement_campaign_context_processor(client):
+def test_active_announcement_campaign_context_processor(client, empty_active_announcements_cache):
     campaign = AnnouncementCampaignFactory(with_item=True)
 
     response = client.get(reverse("search:employers_home"))
diff --git a/tests/www/news/test_views.py b/tests/www/news/test_views.py
index efc445542..ac9289ac1 100644
--- a/tests/www/news/test_views.py
+++ b/tests/www/news/test_views.py
@@ -12,8 +12,11 @@ from tests.utils.test import parse_response_to_soup, reset_model_sequence
 reset_announcement_campaign = pytest.fixture(reset_model_sequence(AnnouncementCampaign))
 
 
-@pytest.mark.empty_active_announcements_cache
 class TestNewsRender:
+    @pytest.fixture(autouse=True)
+    def empty_announcements_cache(self, empty_active_announcements_cache):
+        pass
+
     def test_campaign_rendered_dashboard_prescripteur(self, client, snapshot, reset_announcement_campaign):
         campaign = AnnouncementCampaignFactory(for_snapshot=True)
 
```

Reply:
J'aime que ta solution évite ajouter un `mark` assez niche, merci 🙏